### PR TITLE
fix: deleting access token cache for marketo bulk upload destination

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

We are removing the access token cache to avoid setting faulty TTL for bulk upload.

Write a brief explainer on your code changes.

While load testing with 1M events we could figure out that, as transformer repository is not having a centralised cache, hence if multiple transformer pods are in working state, they will set up authCache for 1 hr, starting at multiple times.

The problem with marketo is, every time ( read it as from every pod ) when we are fetching access token, it is generating the same access token and only the validity time time changes.

For example: 
This is how the chronology of situation takes place.

1. pod 1 --> generates token at --> 11:00 AM and token is "dummyABC", expiry time : 3000s
       a. at this point pod 2 is having the older access token only

2. pod 2 --> generates token at --> 11:10 AM and token is the same "dummyABC", expiry time : 2400s

3. But according to our present auth cache system, in pod 1 the token will be valid till 12:00 AM  and in pod 2 the token will be valid till 12:10 AM
4. pod 1 --> refreshes token at 12:00 PM
5. pod 2 --> is scheduled to refresh at 12:10 PM

which means we have no way to avoid some event failure for pod 2, as the access token is already expired and pod will not refresh it unless some events fail.

For the above scenario, as a quick fix we are dropping authCache implementation for now. As this is a bulk upload destination, we hope we do not exhaust our rate limits ( on the basis of 1M sync, we have not come across any ) 

## What is the related Linear task?

Resolves INT-1404

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

yes.
Presently we are caching the access tokens.
After this change we will make axios call each time for upload, poll and fetch job status.

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3029/test-report.html

File: src/v0/destinations/marketo_bulk_upload/util.js [hunk 1]
```diff
@@ -19,12 +19,12 @@ const {
   FILE_UPLOAD_ERR_MSG,
   ACCESS_TOKEN_FETCH_ERR_MSG,
 } = require('./config');
-const Cache = require('../../util/cache');
+// const Cache = require('../../util/cache');
```
Comment (chandumlg) [inline src/v0/destinations/marketo_bulk_upload/util.js:22]: Remove the comments

File: src/v0/destinations/marketo_bulk_upload/util.js [hunk 2]
```diff
@@ -142,54 +142,48 @@ const getAccessTokenURL = (config) => {
...
 // Fetch access token from client id and client secret
 // DOC: https://developers.marketo.com/rest-api/authentication/
-const getAccessToken = async (config) =>
-  authCache.get(getAccessTokenCacheKey(config), async () => {
-    const url = getAccessTokenURL(config);
-    const { processedResponse: accessTokenResponse } = await handleHttpRequest('get', url, {
-      destType: 'marketo_bulk_upload',
-      feature: 'transformation',
-    });
+const getAccessToken = async (config) => {
```
Comment (chandumlg) [inline src/v0/destinations/marketo_bulk_upload/util.js:145]: Can you please write tests for all the scenarios in this function?

File: src/v0/destinations/marketo_bulk_upload/util.js (hunk 1 above)
Comment (shrouti1507) [reply src/v0/destinations/marketo_bulk_upload/util.js]: removed

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3029/test-report.html

File: test/__tests__/data/marketo_bulk_upload_jobStatus_output.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
       },
       {
         "statusCode": 400,
-        "error": "Unable to fetch job status: due to error "
+        "error": "Unable to fetch job status: due to error \"\""
```
Comment (chandumlg) [inline test/__tests__/data/marketo_bulk_upload_jobStatus_output.json:11]: Can we improve the mock(?) to give a valid error?

File: src/v0/destinations/marketo_bulk_upload/util.js (hunk 2 above)
Comment (shrouti1507) [reply src/v0/destinations/marketo_bulk_upload/util.js:126]: added

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3029/test-report.html

File: src/v0/destinations/marketo_bulk_upload/util.js [hunk 4]
```diff
@@ -142,54 +123,51 @@ const getAccessTokenURL = (config) => {
...
-        accessTokenResponse,
-        ACCESS_TOKEN_FETCH_ERR_MSG,
-        FETCH_ACCESS_TOKEN,
-        config,
-      );
-    }
+  // sample response : {response: '[ENOTFOUND] :: DNS lookup failed', status: 400}
+  if (
+    isDefinedAndNotNull(accessTokenResponse.status) &&
+    !isHttpStatusSuccess(accessTokenResponse.status)
```
Comment (utsabc) [inline src/v0/destinations/marketo_bulk_upload/util.js:136]:
```suggestion
  if (!isHttpStatusSuccess(accessTokenResponse.status)
```

File: src/v0/destinations/marketo_bulk_upload/util.js [hunk 5]
```diff
@@ -142,54 +123,51 @@ const getAccessTokenURL = (config) => {
...
-    throw new AbortedError(
-      `Could not retrieve authorisation token due to error ${accessTokenResponse}`,
-      400,
-    );
-  });
+    return accessTokenResponse.response.access_token;
+  }
+  throw new AbortedError(
+    `Could not retrieve authorisation token due to error ${JSON.stringify(accessTokenResponse)}`,
+    400,
```
Comment (utsabc) [inline src/v0/destinations/marketo_bulk_upload/util.js:168]:
```suggestion
    500,
```

File: src/v0/destinations/marketo_bulk_upload/util.js [hunk 6]
```diff
@@ -142,54 +123,51 @@ const getAccessTokenURL = (config) => {
...
+      );
     }
-    throw new AbortedError(
-      `Could not retrieve authorisation token due to error ${accessTokenResponse}`,
-      400,
-    );
-  });
+    return accessTokenResponse.response.access_token;
+  }
+  throw new AbortedError(
```
Comment (utsabc) [inline src/v0/destinations/marketo_bulk_upload/util.js:166]:
```suggestion
  throw new RetyrableError(
```

**Review by utsabc — APPROVED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3029/test-report.html

**Review by yashasvibajpai — APPROVED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3029/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3029/test-report.html

File: src/v0/destinations/marketo_bulk_upload/util.js [hunk 7]
```diff
@@ -75,29 +66,19 @@ const getAccessTokenCacheKey = (config = {}) => {
  *   console.log(error);
  * }
  */
-const handleCommonErrorResponse = (apiCallResult, OpErrorMessage, OpActivity, config) => {
+const handleCommonErrorResponse = (apiCallResult, OpErrorMessage, OpActivity) => {
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply src/v0/destinations/marketo_bulk_upload/util.js]: addressed

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3029/test-report.html

**Pull request merged**